### PR TITLE
Fix issue where pager would be set to `more` or `most`

### PR DESCRIPTION
**Pull request by ippsav**
### Bug
Colors wouldn't render properly in case the `$PAGER` is set to `more` or `most`.(fix for #1490 )
20230729_23h00m14s_grim
### Solution
Handle the case where `$PAGER` is set to `more` or `most` by using `bat::config::get_pager_executable` which sets the pager to `less` if that's the case.

Unit tests:

> env.rs:
>   - test_env_parsing_with_pager_set_to_bat
>   - test_env_parsing_with_pager_set_to_more
>   - test_env_parsing_with_pager_set_to_most

### CODE REVIEW

File: src/env.rs [hunk 1]
```diff
@@ -41,7 +41,11 @@ impl DeltaEnv {
         let pagers = (
             env::var(DELTA_PAGER).ok(),
             env::var(BAT_PAGER).ok(),
```
Comment (dandavison) [inline src/env.rs:43]: Would you mind getting rid of all explicit mention of `BAT_PAGER`. `get_pager_executable` will look at `BAT_PAGER` itself.

File: src/env.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@ const DELTA_EXPERIMENTAL_MAX_LINE_DISTANCE_FOR_NAIVELY_PAIRED_LINES: &str =
     "DELTA_EXPERIMENTAL_MAX_LINE_DISTANCE_FOR_NAIVELY_PAIRED_LINES";
 const DELTA_PAGER: &str = "DELTA_PAGER";
 const BAT_PAGER: &str = "BAT_PAGER";
-const PAGER: &str = "PAGER";
+// const PAGER: &str = "PAGER";
```
Comment (dandavison) [inline src/env.rs:13]: Could you remove this commented line please.

File: src/env.rs [hunk 3]
```diff
@@ -41,7 +41,11 @@ impl DeltaEnv {
         let pagers = (
             env::var(DELTA_PAGER).ok(),
             env::var(BAT_PAGER).ok(),
-            env::var(PAGER).ok(),
+            // We're using `bar::config::get_pager_executable` here instead of just returning
```
Comment (dandavison) [inline src/env.rs:44]:
```suggestion
            // We're using `bat::config::get_pager_executable` here instead of just returning
```

**Review by dandavison — CHANGES_REQUESTED** [review]
This is awesome, thanks very much! I've made a couple of small requests for changes to the diff.

**Comment by ippsav** [comment]
@dandavison all fixed ! that's true now that i read the source code of bat, no need for handling `BAT_PAGER`, thanks !

**Comment by dandavison** [comment]
Perfect, thanks for doing this!

**Comment by ippsav** [comment]
You're welcome ! it was a fun thing to do ! and thanks for making such a useful app !

**Pull request merged**